jvb/src/main/java/org/jitsi/videobridge/Videobridge.java

```diff
@@ -255,18 +264,23 @@ public Videobridge(
     /**
      * Enables graceful shutdown mode on this bridge instance and eventually
      * starts the shutdown immediately if no conferences are currently being
-     * hosted. Otherwise bridge will shutdown once all conferences expire.
+     * hosted. Otherwise, the bridge will shutdown once all conferences expire.
      */
     private void enableGracefulShutdownMode()
     {
-        if (!shutdownInProgress)
+        if (shutdownRequestedTime == null)
         {
             logger.info("Entered graceful shutdown mode");
         }
-        this.shutdownInProgress = true;
+        this.shutdownRequestedTime = clock.instant();
```

Comment:
Hm, just noticed this - if enableGracefulShutdownMode is called more than once, this reset the shutdown time.  Is that what we want?